### DISCUSSION TITLE
Support sphinx doc strings

### DISCUSSION THREAD

**Discussion by rchiodo** [microsoft/pylance-release#4416] [Enhancement]
@jakebailey I've created a new issue for an actual bug related to missing newlines for "key" and "keyword" parameter descriptions. These are valid syntax and break the current implementation of the docstring hover for sphinx format: https://github.com/microsoft/pylance-release/issues/2002

---

For the sake of not making new duplicate tickets, I would like to continue the discussion that is exactly related to "The Sphinx docstring hover may be more pretty". The Sphinx website shows an example rendering of how their syntax is meant to be used: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists

This is the expected rendering, which would bring it to parity with the same feature in PyCharm and match Sphinx's documentation.
![image](https://user-images.githubusercontent.com/88157114/139322820-d09c440f-7353-4f8f-badc-e1e337169d62.png)

_Originally posted by @jeremyvital4 in https://github.com/microsoft/pylance-release/issues/1113#issuecomment-954139426_

**Comment by rchiodo** [comment]
That comment had 21 upvotes before I created this.

**Comment by debonte** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by TurnrDev** [comment]
I too would love to see some proper docstring support in VSCode, it's hideous looking at unparsed strings

**Comment by StellaHuang95** [comment]
Another similar request from user: https://github.com/microsoft/pylance-release/issues/4445

**Comment by dill0wn** [comment]
Like most of Python, sphinx provided a feature that the python spec did not handle. Many older codebases use sphinx (as well as most(all?) readthedocs sites) to provide docstrings. 

Are there any docstring comment formats that are formatted/rendered by pylance?

**Reply by rchiodo** [reply]
We support a number of formats but they're not rendered to the extent described above. The list we currently have is:

- Epytext
- reST
- Google doc strings

Mostly the formatting we do is to make sure line feeds are correct and spacing is consistent.

**Comment by rchiodo** [comment]
This should be implemented now under the `python.analysis.supportRestructuredText` option.
